src/libs/conduit/conduit_node.cpp

```diff
@@ -11981,6 +11981,341 @@ Node::to_string_default() const
+    }
+
+    if(opts.has_child("depth") &&
+       opts["depth"].dtype().is_number())
+    {
+       depth = (index_t)opts["depth"].to_int32();
+    }
+
+    if(opts.has_child("pad") &&
+       opts["pad"].dtype().is_string())
+    {
+       pad = opts["pad"].as_string();
+    }
+
+    if(opts.has_child("eoe") &&
+       opts["eoe"].dtype().is_string())
+    {
+       eoe = opts["eoe"].as_string();
+    }
+
+    to_summary_string_stream(os,
+                             num_children_threshold,
+                             num_elements_threshold,
+                             indent,
+                             depth,
+                             pad,
+                             eoe);
+}
+
+
+//-----------------------------------------------------------------------------
+//-- (private interface)
+//-----------------------------------------------------------------------------
+void
+Node::to_summary_string_stream(const std::string &stream_path,
+                               const conduit::Node &opts) const
+{
+    std::ofstream ofs;
+    ofs.open(stream_path.c_str());
+    if(!ofs.is_open())
+    {
+        CONDUIT_ERROR("<Node::to_summary_string_stream> failed to open file: "
+                      << "\"" << stream_path << "\"");
+    }
+    to_summary_string_stream(ofs,opts);
+    ofs.close();
+}
+
+//-----------------------------------------------------------------------------
+std::string
+Node::to_summary_string_default() const
+{
+    return to_summary_string();
+}
+
+
+//-----------------------------------------------------------------------------
+//-- (private interface)
+//-----------------------------------------------------------------------------
+void
+Node::to_summary_string_stream(std::ostream &os,
+                               index_t num_children_threshold,
+                               index_t num_elements_threshold,
+                               index_t indent,
+                               index_t depth,
+                               const std::string &pad,
+                               const std::string &eoe) const
+{
+    // rubber, say hello to the road:
+
+    std::ios_base::fmtflags prev_stream_flags(os.flags());
+    os.precision(15);
+    if(dtype().id() == DataType::OBJECT_ID)
+    {
+        os << eoe;
+        int nchildren = m_children.size();
+        int threshold = num_children_threshold;
+
+        // if we are neg or zero, show all children
+        if(threshold <=0)
+        {
+           threshold = nchildren;
+        }
+
+        // if above threshold only show threshold # of values
+        int half = threshold / 2;
+        int bottom = half;
+        int top = half;
+        int num_skipped = m_children.size() - threshold;
+
+        //
+        // if odd, show 1/2 +1 first
+        //
+
+        if( (threshold % 2) > 0)
+        {
+            bottom++;
+        }
+
+        bool done  = (nchildren == 0);
+        int idx = 0;
+
+        while(!done)
```

Comment:
The code in these loops is rather hard for me to follow. I think using the lambdas we're afforded by C++11 could make it much more legible, e.g.:

```
const auto to_summary = [] (const conduit::Node& n, ...)
{
    // ... //
}

for(index_t idx = 0; idx <= bottom; idx++)
{
    to_summary(node);
}

// ... skipped print logic goes here ... //

for(index_t idx = nchildren - top; idx < nchildren; idx++)
{
    to_summary(node);
}
```

Just my 2 cents here; I don't mind keeping this as is for the sake of expediency.

Reply:
added https://github.com/LLNL/conduit/issues/699 to track this.

Reply:
(i think this is great idea)